src/components/TextInput.tsx

```diff
@@ -0,0 +1,160 @@
+    prefixIcon?: IconOption;
+    suffixIcon?: IconOption;
+    validationState?: "error" | "success" | null;
+    autoIcons?: boolean;
+}
+
+/**
+ * Features and Props:
+ * 
+ * - Supports validation states such as error and success.
+ * - Allows toggling of hidden password option.
+ * - Ability to set the input field as disabled.
+ * - Option to include prefix and suffix icons.
+ * - Supports popover info icon.
+ * - Placeholder text can be customized.
+ * - Provides value and onChange handler for input control.
+ * 
+ * Default Animations:
+ * 
+ * - Focus animation includes border color change and slight shadow effect.
+ * - Shake animation triggered on validation error.
+ * 
+ * @param {string} value - The current value of the input field.
+ * @param {Function} onChange - The function to handle changes in the input field.
+ * @param {string} placeholderText - The text to display as a placeholder in the input field.
+ * @param {boolean} isPassword - Indicates if the input field should be of type password.
+ * @param {boolean} isDisabled - Indicates if the input field should be disabled.
+ * @param {string} prefixIcon - The icon to display as a prefix in the input field.
+ * @param {string} suffixIcon - The icon to display as a suffix in the input field.
+ * @param {"error" | "success" | null } validationState - The validation state of the input field.
+ * 
+ * @returns {JSX.Element} 
+ */
+export const TextInput = ({
+    value,
+    onChange,
+    placeholderText,
+    isPassword,
+    isDisabled,
+    prefixIcon,
+    suffixIcon,
+    validationState = null,
+    autoIcons = true
+}: TextInputProps) => {
+
+
+
+    // Let's show some default icons if autoIcons is true
+    if (autoIcons && !prefixIcon && isPassword) {
+        prefixIcon = "lock"
+    }
+    if (autoIcons && !suffixIcon && validationState === "success") {
+        suffixIcon = "check"
+    }
+    if (autoIcons && !suffixIcon && validationState === "error") {
+        suffixIcon = "error"
+    }
+
+    const PrefixIconObject: IconType | null = iconLookup(prefixIcon)
+
+    const SuffixIconObject: IconType | null = iconLookup(suffixIcon)
+
+
+
+    const containerStyle = `
+        relative 
+        flex 
+        flex-col 
+        flex-grow 
+        rounded-lg 
+        m-2 
+
+    `;
+
+    // Padding notes:
+    // w-6 + m-3 + m-3 = *-12 (48px) on the icon
+    // this must equal the pl-12 or pr-12 padding used in inputPadding box when icon appears
+    // 
+    // Transform notes:
+    // top-1/2 puts top edge at 50% of containing element's height
+    // transform -translate-y-1/2 ...translates element vertically by -50% of its own height
+    const iconStyle = "h-6 w-6 mx-3 top-1/2 transform -translate-y-1/2";
+
+    const inputPadding = `
+        rounded-lg 
+        p-3 
+        ${prefixIcon ? "pl-12" : ""} 
+        ${suffixIcon ? "pr-12" : ""}
+        border border-2
+        ${validationState === "success" ? "border-green-500" : ""} 
```

Comment:
nit: consider using the classnames package if it makes this much more readable, but don't fixate on it if it's confusing. It's pretty simple though.

https://www.npmjs.com/package/classnames